cpp/src/arrow/compute/kernels/aggregate_pivot.cc

```diff
@@ -42,47 +44,56 @@ struct PivotImpl : public ScalarAggregator {
       values_.push_back(MakeNullScalar(value_type));
     }
     out_type_ = struct_(std::move(fields));
-    ARROW_ASSIGN_OR_RAISE(key_mapper_, PivotWiderKeyMapper::Make(*key_type_, options_));
+    ARROW_ASSIGN_OR_RAISE(key_mapper_,
+                          PivotWiderKeyMapper::Make(*key_type_, options_, ctx));
     return Status::OK();
   }
 
   Status Consume(KernelContext*, const ExecSpan& batch) override {
     DCHECK_EQ(batch.num_values(), 2);
     if (batch[0].is_array()) {
-      ARROW_ASSIGN_OR_RAISE(span<const PivotWiderKeyIndex> keys,
-                            key_mapper_->MapKeys(batch[0].array));
+      ARROW_ASSIGN_OR_RAISE(auto keys_array, key_mapper_->MapKeys(batch[0].array));
```

Comment:
Shall we assert the `keys_array`'s data type?

Reply:
Will do.